src/protocol/V1.php

```diff
@@ -18,21 +18,21 @@ class V1 extends AProtocol
 {
 
     /**
+     * Send INIT message
      * @param mixed ...$args
      * @return array
      * @throws Exception
      */
     public function init(...$args): array
     {
-        if (count($args) < 4) {
+        if (count($args) < 1) {
             throw new PackException('Wrong arguments count');
         }
 
-        $this->write($this->packer->pack(0x01, $args[0], [
-            'scheme' => $args[1],
-            'principal' => $args[2],
-            'credentials' => $args[3]
-        ]));
+        $userAgent = $args[0]['user_agent'];
+        unset($args[0]['user_agent']);
```

Comment:
bolt v1 had user agent as separated message argument, we have to extract it from extra parameter.